File: src/contests/cg/puzzle/pascal/1d-bush-fire.pas
// 🎮 CodinGame Puzzle - 1d-bush-fire
// https://www.codingame.com/training/easy/1d-bush-fire

{$mode objfpc}{$H+}
var
  n, i, drops, j: integer;
  strip: string;
begin
  readln(n);
  for i := 1 to n do
  begin
    readln(strip);
    drops := 0;
    j := 1;
    while j <= length(strip) do
    begin
      if strip[j] = 'f' then
      begin
        // Drop at j covers j, j+1, j+2 — skip past all 3.
        inc(drops);
        j := j + 3;
      end
      else
        inc(j);
    end;
    writeln(drops);
  end;
end.

File: src/contests/cg/puzzle/clojure/1d-bush-fire.clj
;; 🎮 CodinGame Puzzle - 1d-bush-fire
;; https://www.codingame.com/training/easy/1d-bush-fire

(ns Solution (:gen-class))

(defn -main [& _]
  (let [n (Integer/parseInt (.trim (read-line)))]
    (dotimes [_ n]
      (let [strip (.trim (read-line))
            len (count strip)]
        (loop [j 0 drops 0]
          (if (< j len)
            (if (= (.charAt strip j) \f)
              ;; Drop at j covers j, j+1, j+2 — skip past all 3.
              (recur (+ j 3) (inc drops))
              (recur (inc j) drops))
            (println drops))))))
  (flush))
